model/job.go

```diff
@@ -295,21 +307,24 @@ func (j *Job) WriteConfigs(role *Role, lightOpinionsPath, darkOpinionsPath strin
 		Networks   struct {
 			Default map[string]string `json:"default"`
 		} `json:"networks"`
-		ExportedProperties []string                    `json:"exported_properties"`
-		Consumes           map[string]*JobLinkConsumes `json:"consumes"`
+		ExportedProperties []string               `json:"exported_properties"`
+		Consumes           map[string]jobLinkInfo `json:"consumes"`
 	}
 
 	config.Job.Templates = make([]jobConfigTemplate, 0)
 	config.Parameters = make(map[string]string)
 	config.Properties = make(map[string]interface{})
 	config.Networks.Default = make(map[string]string)
 	config.ExportedProperties = make([]string, 0)
-	config.Consumes = make(map[string]*JobLinkConsumes, 0)
+	config.Consumes = make(map[string]jobLinkInfo)
 
 	config.Job.Name = role.Name
 
-	for _, roleJob := range role.Jobs {
+	for _, roleJob := range role.RoleJobs {
 		config.Job.Templates = append(config.Job.Templates, jobConfigTemplate{roleJob.Name})
```

Comment:
I know this hasn't been changed, but why are we merging all the templates from all the jobs inside the role containing `j`? Just trying to understand what it does.

Reply:
The templates are, as far aw I know, completely useless. Nobody actually looks at it.

Since this is `Job.WriteConfigs()`, we don't know which roleJob we're touching, though. So we still need to get the resolved consumers from there… I guess I can filter by checking if `roleJob.Job` is `j`?